### PR TITLE
Workaround for UWP bug in HttpStack when cloning the http con…

### PR DESCRIPTION
**Pull request by bgavrilMS**
…tent

### CODE REVIEW

File: core/src/Http/HttpRequest.cs [hunk 1]
```diff
@@ -236,5 +238,32 @@ private static async Task<HttpResponse> CreateResponseAsync(HttpResponseMessage
...
+                }
+            }
+
+            // WORKAROUND 
+            // On UWP there is a bug in the Http stack that causes an exception to be thrown when moving around a stream.
+            // https://stackoverflow.com/questions/31774058/postasync-throwing-irandomaccessstream-error-when-targeting-windows-10-uwp
+            // LoadIntoBufferAsync is necessary to buffer content for multiple reads - see https://stackoverflow.com/questions/26942514/multiple-calls-to-httpcontent-readasasync
+            // Documentation is sparse, but it looks like loading the buffer into memory avoids the bug, without 
+            // replacing the System.Net.HttpClient with Windows.Web.Http.HttpClient, which is not exactly a drop in replacement
+            await clone.LoadIntoBufferAsync().ConfigureAwait(false);
```
Comment (bgavrilMS) [inline core/src/Http/HttpRequest.cs:264]: This line is the actual fix

File: core/src/Helpers/CoreHelpers.cs [hunk 2]
```diff
@@ -233,20 +233,5 @@ private static void AddKeyValueString(StringBuilder messageBuilder, string key,
             messageBuilder.Append(value);
         }
 
-        internal static async Task<HttpContent> DeepCopyAsync(HttpContent content)
```
Comment (bgavrilMS) [inline core/src/Helpers/CoreHelpers.cs:236]: Not needed anywhere else but in HttpRequest below.

**Review by jennyf19 — APPROVED** [review]
LGTM

File: core/src/Http/HttpRequest.cs (hunk 1 above)
Comment (henrik-me) [reply core/src/Http/HttpRequest.cs:264]: nice find! should there be an #if UWP for this then?

---
In reply to: [220696902](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1273#discussion_r220696902) [](ancestors = 220696902)

File: adal/devApps/AdalUAPTestApp/MainPage.xaml.cs [hunk 3]
```diff
@@ -164,5 +161,41 @@ private async void AccessTokenSilentButton_Click(object sender, RoutedEventArgs
...
+        {
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+
+            this.AccessToken.Text = string.Empty;
+            AuthenticationContext context = new AuthenticationContext("https://login.microsoftonline.com/common");
+            try
+            {
+                AuthenticationResult authResult = await context.AcquireTokenAsync(
+                    "https://graph.microsoft.com",
+                    ClientId, new UserCredential("bogavril@microsoft.com"))
```
Comment (henrik-me) [inline adal/devApps/AdalUAPTestApp/MainPage.xaml.cs:178]: >"bogavril@microsoft.com") [](start = 49, length = 25)

should this be an input via the UI instead? (perhaps the UI should be grabbing this from the user of the system by default?)

**Comment by henrik-me** [comment]
Need any updates for MSAL apps?

**Comment by bgavrilMS** [comment]
@henrik-me - I made an update so that the app fetches the username from the OS.

**Comment by henrik-me** [comment]
thanks

File: core/src/Http/HttpRequest.cs (hunk 1 above)
Comment (bgavrilMS) [reply core/src/Http/HttpRequest.cs:264]: Yes, I guess so. I'll address this in a separate PR.

**Pull request merged**